### ISSUE TITLE
Make `postcss-safe-parser` as peer dependency

### ISSUE THREAD

**Issue by ramram1048** [stylelint/postcss-css-in-js#212]
# TL; DR: Make `postcss-safe-parser` as peer dependencies.

Hi, I'm a huge fan of stylelint. Currently I'm using stylelint with `VS Code`, `Yarn Berry (PnP)` and `css-in-js`. 
When I was trying to write some styled component with props, I got error message like this:

```
Error: @stylelint/postcss-css-in-js tried to access postcss-safe-parser, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.  

Required package: postcss-safe-parser (via "postcss-safe-parser/lib/safe-parser") 
Required by: @stylelint/postcss-css-in-js@virtual:f63b767..2722fdc#npm:0.37.2 (via ~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/)  
Require stack: - ~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/template-safe-parser.js 
- ~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/template-safe-parse.js 
- ~/.yarn/__virtual__/postcss-syntax-virtual-9d9712c9f9/0/cache/postcss-syntax-npm-0.36.2-ef243fe3b7-812baee602.zip/node_modules/postcss-syntax/load-syntax.js 
- ~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/extract.js 
- ~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/index.js 
- ~/.yarn/cache/stylelint-npm-13.13.1-f63b76732e-9dafa8d90f.zip/node_modules/stylelint/lib/syntaxes/syntax-css-in-js.js 
- ~/.yarn/cache/stylelint-npm-13.13.1-f63b76732e-9dafa8d90f.zip/node_modules/stylelint/lib/syntaxes/index.js 
- ~/.yarn/cache/stylelint-npm-13.13.1-f63b76732e-9dafa8d90f.zip/node_modules/stylelint/lib/getPostcssResult.js 
- ~/.yarn/cache/stylelint-npm-13.13.1-f63b76732e-9dafa8d90f.zip/node_modules/stylelint/lib/createStylelint.js 
- ~/.yarn/cache/stylelint-npm-13.13.1-f63b76732e-9dafa8d90f.zip/node_modules/stylelint/lib/index.js 
- ~/.pnp.cjs
     at Function.external_module_.Module._resolveFilename (~/.pnp.cjs:22506:55)
     at Function.external_module_.Module._load (~/.pnp.cjs:22305:48)
     at Module.require (internal/modules/cjs/loader.js:952:19)     at require (internal/modules/cjs/helpers.js:88:18)
     at Object.<anonymous> (~/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-717c294e8b/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/template-safe-parser.js:5:20)
     at Module._compile (internal/modules/cjs/loader.js:1063:30)
     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
     at Module.load (internal/modules/cjs/loader.js:928:32)
     at Function.external_module_.Module._load (~/.pnp.cjs:22355:14)
     at Module.require (internal/modules/cjs/loader.js:952:19)
```

The source was:

```tsx
import styled from "@emotion/styled";

export const ContentWrap = styled.div`
  padding: 20px;
  background-color: ${(props) => props.theme.color.primary};
`;
```

The error was only came with VS Code Stylelint extension, not in CLI, but it was very annoyed for me. Anyway it resolved that just tell yarn to use `postcss-safe-parser` as peer dependency like this:

 ```yml
# .yarnrc.yml

# ... other settings ...

packageExtensions:
  # ... other resolutions ...
  "@stylelint/postcss-css-in-js@^0.37.2":
    peerDependencies:
      postcss-safe-parser: "^4.0.2"  # same with dev dependency version of postcss-css-in-js@0.37.2
 ```

And here, I think that this resolution could be marked in this package, not in every other's package. As I'm just beginner of JavaScript and open sources, I don't know this will be reasonable or not. If it'll be okay, I'm willing to create PR of this!

Maybe related: https://github.com/stylelint/stylelint/issues/4793#issuecomment-706654896
